az-cvm-vtpm/hcl/src/lib.rs

```diff
@@ -1,21 +1,30 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-use crate::tdx::TdxVmReport;
 use jsonwebkey::JsonWebKey;
 use memoffset::offset_of;
 use serde::{Deserialize, Serialize};
 use serde_big_array::BigArray;
-use sev::firmware::guest::AttestationReport as SnpVmReport;
+use sev::firmware::guest::AttestationReport as SnpReport;
 use sha2::{Digest, Sha256};
-use std::mem;
+use std::convert::TryFrom;
+use std::mem::size_of;
+use std::ops::RangeInclusive;
+use tdx::TdReport;
 use thiserror::Error;
 
 const HCL_AKPUB_KEY_ID: &str = "HCLAkPub";
-const MAX_REPORT_SIZE: usize = mem::size_of::<SnpVmReport>();
-const MIN_REPORT_SIZE: usize = mem::size_of::<TdxVmReport>();
+const TD_REPORT_SIZE: usize = size_of::<TdReport>();
+const SNP_REPORT_SIZE: usize = size_of::<SnpReport>();
+const MAX_REPORT_SIZE: usize = SNP_REPORT_SIZE;
 const SNP_REPORT_TYPE: u32 = 2;
 const TDX_REPORT_TYPE: u32 = 4;
+const HW_REPORT_OFFSET: usize = offset_of!(AttestationReport, hw_report);
+const fn report_range(report_size: usize) -> RangeInclusive<usize> {
+    HW_REPORT_OFFSET..=(HW_REPORT_OFFSET + report_size)
```

Comment:
what is `..=`?

Reply:
it's a range expression, albeit the wrong one. good catch.